rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -182,11 +182,16 @@ object Runtime {
             freeCount = arity
           )
         )
-        val continuation                   = TaggedContinuation(ScalaBodyRef(ref))
-        implicit val MATCH_UNLIMITED_PHLOS = matchListPar(Cost(Integer.MAX_VALUE))
+        val continuation = TaggedContinuation(ScalaBodyRef(ref))
+        val cost0: _cost[F] =
```

Comment:
So, I did this because I was trying to preserve the original semantics, for some reason thinking reading `MATCH_UNLIMITED_PHLOS` as a `def` and not a `val`. As a `def`, it would provide both `space.install` and `replaySpace.install` with maximum gas. Now, I'm realizing that we don't even need to instantiate new cost accounts with maximum gas. 

Lukasz. A special cost type would require some way to indicate when you're entering and exiting its usage. So, maybe a bracket operation like `def free[A, B](fa: A => F[B]): F[B]` that doesn't charge for computing `fa` would suffice.

To Artur's point, I agree. Ideally, we shouldn't need to provide `Cost(Integer.MAX_VALUE)` to simulate free operations. We should be able to just not charge in the code and pass `_cost[F]` implicitly. The only problem with doing this for `install` is that it eventually calls `m.get(pattern, matchCandidate)`, which costs something. I'll have to think on this for a longer term solution.

I'll revise to de-duplicate the cost accounts and change `Cost(Integer.MAX_VALUE)` to `Cost.UNSAFE_UNLIMITED`.